README.md

```diff
@@ -25,3 +25,20 @@ The app deployed as part of this solution does the following:
 
 The agent crashes whenever it encounters an unrecoverable error with the metadata APIs.
 This agent is not production hardened yet and so use it with caution.
+
+## Graceful terminations for regular pods (Non-system pods)
+
+The pods that are not in the kube-system are called **regular pods** in this agent.
+By default, regular pods are deleted immediately before deleting system pods.
+If you want to delete regular pods gracefully, please add `--system-pod-grace-period=n` in arguments according to the following rules:
+
+- If targeted VM is Preemptible VM, specify `n` with a value from `0s` to `14s`.
```

Comment:
Based on the code:
```
if timeout >= 2*p.systemPodGracePeriod {
    gracePeriod = int64(timeout.Seconds() - p.systemPodGracePeriod.Seconds())
}
```

Assuming timeout == 30 seconds (for Preemptible VM), user can pass systemPodGracePeriod from 0 to 15s to ensure grace period for regular pods is greater than 0. It's 15 because check is >= and >.

It might be useful to note that if user passes 0 for system pods, the system pods will be terminated immediately and regular pods will have 30 seconds of grace period. If users pass 15 seconds, both system pods and regular pods will have 15 seconds grace period both. This is because termination is done in sequential matter, regular pods first, then system pods.

Reply:
Sounds good.
It will be very helpful to indicate the actual scenario.

However, the conditional expression is certainly equal or greater than, but as I said above, we can't allocate all 30 seconds to this. And in that case, specifying 15 seconds for `systemGracePeriod` should not meet this condition.
It's very simple, but here's sample code for what actually happens in this library:

```
package main

import (
	"fmt"
	"time"
)

func main() {
	// Asssuming this works like `TerminationTime`.
	t := time.Now().Add(30 * time.Second)
	// Assuming this works like `timeout`
	t2 := t.Sub(time.Now())
	fmt.Println(t2.Seconds()) // 29.999995477
}
```

Reply:
Thanks, I see what you mean, by the time it hits the check it should be less than 15 seconds since it takes at least a bit of time to get to that check.